### ISSUE TITLE
Make secretKeyRef optional

### ISSUE THREAD

**Issue by hodor9** [newrelic-experimental/monitoring-kubernetes-with-opentelemetry#11]
For our use case for example we have like 70 customer, which would have like 140 APIkeys (secrets). We want to rollout nr-otel globally on all clusters as an addon as a part of our infrastructure, so we have same versioning, same setup across all clusters.

Not all our customers use all our services, for example if we have order management system hosted on specific clusters, only 6 of our customers use our order management system, so from a security point of view, i don't want to roll out all customer secrets on that cluster. Therefore I can think of 2 solutions to that problem, but maybe there are more and better solutions.
We are going to fix the problem with the secrets by running a CronJob, that will iterate through namespaces on the cluster, check which customers are present and get their corresponding API Keys from Azure KeyVault and roll them out as a secret on the cluster. With present implementation the daemonset/statefulset or deployment collectors require all secrets to be present, that are defined in the global config, which means that their pods won't start until those secrets are present.

- An easy Solution would be to define the secretKeyRefs as optional. That way if it is not present the Pod will still start with the secrets that exist (for those 6 customers from the example above)

Lets say we have for example:

```
global:
  newrelic:
    enabled: true
    endpoint: "OTLP_ENDPOINT_FOR_ALL_ACCOUNTS"
    teams:
      devteam1:
        licenseKey:
          secretRef:
            name: "<YOUR_EXISTING_SECRET_DEVTEAM_1>"
            key: "<KEY_TO_LICENSE_KEY_WITHIN_THE_SECRET_DEVTEAM_1>"
        namespaces: []
      devteam2:
        licenseKey:
          secretRef:
            name: "<YOUR_EXISTING_SECRET_DEVTEAM_2>"
            key: "<KEY_TO_LICENSE_KEY_WITHIN_THE_SECRET_DEVTEAM_2>"
        namespaces:
          - namespace-devteam-2
   ............
      devteam70:
        licenseKey:
          secretRef:
            name: "<YOUR_EXISTING_SECRET_DEVTEAM_70>"
            key: "<KEY_TO_LICENSE_KEY_WITHIN_THE_SECRET_DEVTEAM_70>"
        namespaces:
          - namespace-devteam-70
```

In that use case if we only roll out the secrets for devteam1-6 the deployment will fail, because the other 64 secretrefkeys are expected (so this would need to be optional). For example:

```
    valueFrom:
      secretKeyRef:
        name: <secrets name>
        key: <secrets key>
        optional: true
```

**Comment by utr1903** [comment]
Hey @hodor9, this feature requests makes sense from perspective as well and I will put this enhancement into the next release.

**Comment by utr1903** [comment]
@hodor9 Implemented with #20 👍
